packages/react-ui-ag/src/ToggleDropDown/__tests__/ToggleDropDown-test.js

```diff
@@ -0,0 +1,112 @@
+import ThemedToggleDropDown from '../ToggleDropDown'
+import theme from './mocks/theme'
+
+const ToggleDropDown = ThemedToggleDropDown.WrappedComponent
+
+describe('ToggleDropDown', () => {
+  it('renders the base class and children', () => {
+    const props = {
+      theme,
+      className: theme.ToggleDropDown,
+      children: (<div>Content</div>),
+    }
+
+    const wrapper = mount(<ToggleDropDown {...props} />)
+    expect(wrapper.hasClass('ToggleDropDown')).toBeTruthy()
+    expect(wrapper.find('[data-tid="toggleDropDown-content"]').contains([<div>Content</div>])).toBeTruthy()
+  })
+
+  it('sets the default state to toggled on', () => {
+    const props = {
+      theme,
+      className: theme.ToggleDropDown,
+      children: <a href="#">Toggle</a>,
+    }
+
+    const wrapper = mount(<ToggleDropDown {...props} />)
+    expect(wrapper.find('[data-tid="toggleDropDown-button"]').first().hasClass('ToggleDropDown-Button-on')).toBeTruthy()
+  })
+
+  it('changes switches css class between ToggleDropDown-Button-on and ToggleDropDown-Button-off when toggled', () => {
+    const props = {
+      theme,
+      className: theme.ToggleDropDown,
+      children: <a href="#">Toggle</a>,
+    }
+
+    const wrapper = mount(<ToggleDropDown {...props} />)
+    expect(wrapper.find('[data-tid="toggleDropDown-button"]').first().hasClass('ToggleDropDown-Button-on')).toBeTruthy()
+    wrapper.find('[data-tid="toggleDropDown-button"]').first().simulate('click')
+    expect(wrapper.find('[data-tid="toggleDropDown-button"]').first().hasClass('ToggleDropDown-Button-off')).toBeTruthy()
+  })
+
+  it('sets value to prop value', () => {
+    const props = {
+      theme,
+      className: theme.ToggleDropDown,
+      children: <a href="#">Toggle</a>,
+    }
+
+    const wrapper = mount(<ToggleDropDown {...props} />)
+    expect(wrapper.find('[data-tid="toggleDropDown-button"]').first().hasClass('ToggleDropDown-Button-on')).toBeTruthy()
+    wrapper.setProps({ visible: false })
+    expect(wrapper.find('[data-tid="toggleDropDown-button"]').first().hasClass('ToggleDropDown-Button-off')).toBeTruthy()
+  })
+
+  it('It fires callback function which returns alternating true and false', () => {
+    const props = {
+      theme,
+      className: theme.ToggleDropDown,
+      children: <a href="#">Toggle</a>,
+    }
+    const onClick = jest.fn()
+
+    const wrapper = mount(<ToggleDropDown onClick={onClick} {...props} />)
+    wrapper.find('[data-tid="toggleDropDown-button"]').first().simulate('click')
+    expect(onClick.mock.calls[0][0]).toEqual(false)
+    wrapper.find('[data-tid="toggleDropDown-button"]').first().simulate('click')
+    expect(onClick.mock.calls[1][0]).toEqual(true)
+    wrapper.find('[data-tid="toggleDropDown-button"]').first().simulate('click')
+    expect(onClick.mock.calls[2][0]).toEqual(false)
+  })
+
+  it('changes switches css class between ToggleDropDown-Button-on and ToggleDropDown-Button-off when toggled', () => {
```

Comment:
I think you're accidentally testing this like 3 times 😉  unless I'm missing something